MiniProjectUIKit/ViewController/HomepageViewController.swift

```diff
@@ -0,0 +1,285 @@
+        NSLayoutConstraint.activate([
+            searchBar.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: 8),
+            searchBar.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 8),
+            searchBar.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -8)
+        ])
+        
+        // Gesture recognizer to dismiss keyboard
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
+        tapGesture.cancelsTouchesInView = false
+        view.addGestureRecognizer(tapGesture)
+        
+        // Filter ScrollView
+        filterScrollView.accessibilityIdentifier = "FilterScrollView"
+        filterScrollView.showsHorizontalScrollIndicator = true
+        view.addSubview(filterScrollView)
+        
+        filterScrollView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            filterScrollView.topAnchor.constraint(equalTo: searchBar.bottomAnchor, constant: 8),
+            filterScrollView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 0),
+            filterScrollView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: 0),
+            filterScrollView.heightAnchor.constraint(equalToConstant: 40)
+        ])
+        
+        // Collection View
+        let layout = UICollectionViewFlowLayout()
+        layout.minimumLineSpacing = 16
+        layout.minimumInteritemSpacing = 16
+        layout.sectionInset = UIEdgeInsets(top: 8, left: 16, bottom: 8, right: 16)
+        
+        collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        collectionView.register(RecipeCollectionViewCell.self, forCellWithReuseIdentifier: "RecipeCell")
+        collectionView.backgroundColor = .clear
+        view.addSubview(collectionView)
+        
+        collectionView.accessibilityIdentifier = "RecipeCollectionView"
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            collectionView.topAnchor.constraint(equalTo: filterScrollView.bottomAnchor, constant: 8),
+            collectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            collectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            collectionView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+        
+        // Activity Indicator
+        activityIndicator.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(activityIndicator)
+        NSLayoutConstraint.activate([
+            activityIndicator.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            activityIndicator.centerYAnchor.constraint(equalTo: view.centerYAnchor)
+        ])
+    }
+    
+    // MARK: - Fetch Areas for Filters
+    private func fetchAreas() {
```

Comment:
sebetulnya untuk fetchAreas, ga perlu hit API terpisah, tapi cukup filter unique value `strArea` dari result API yang search

based on requirement -> "Make sure to only show the quick filter that is based on the current search result."

<img width="446" alt="image" src="https://github.com/user-attachments/assets/e3a588e7-ff1f-4501-90c4-3e5924358252" />